operator/docs/enhancements/ruler_support.md

```diff
@@ -0,0 +1,686 @@
+2. Additionally the `LokiRulerConfig** requires a user-provided Kubernetes Secret for sensitive information. This adds an extra dependency that requires validation inside the controller-loop.
+
+### Risks and Mitigations
+
+#### Loki Ruler Configuration versioning
+
+**Risk**: The `LokiRulerConfig` CRD exposes an almost identical [ruler config](https://grafana.com/docs/loki/latest/configuration/#ruler) to support AlertManager and Remote-Write. Both sub-specifications are subject of change to the API version of the server endpoint (i.e. AlertManager, Remote-Write). Although AlertManager is using a versioned approach e.g. `EnableV2` switch, we miss a similar approach accross remote write endpoints.
+
+**Mitigation**: We require a support matrix for `LokiRulerConfig` versions to AlertManager API versions and Remote Write implementations. For later we could use a list of releases (e.g. Prometheus 2.x, Thanos 0.20.z).
+
+#### Single CRD for Alerting and Recording Rules
+
+**Risk**: `LokiRule` is an amalgam of types for alerting and recording rules. This requires custom validation to inform the user of bad inputs and in turn extra maintaince over time. In addition alerting rules might deviate with new features in future (See [Feature Request: alert relabel configs](https://github.com/grafana/loki/issues/5886)). Thus a CRD combining both rules types might become confusing and error-prone from a user experience perspective.
+
+**Mitigation**: Either split the two types in an `AlertingRule` and `RecordingRule` type now or in a future v2 version. Later would probably be harder to support seamless migrations from the Kubernetes API server side.
+
+#### Single ConfigMap for all LokiRule instances
+
+**Risk**: The proposed reconciliation approach manifests that all `LokiRule` instances are transformed to individual entries in a single ConfigMap. Alerting/Recording rules can be become quite big (e.g. large LogQL expresssions). As per ConfigMaps represent Kubernetes resource stored in etcd, this might hit some store limits (See [1MiB limit](https://kubernetes.io/docs/concepts/configuration/configmap/#motivation)). Therefore storing all rules in a single ConfigMap might become impossible over time or on large clusters.
+
+**Mitigation**: Introduce some sort of sharding of LokiRules into multiple ConfigMap resources.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. Do we need to split the `LokiRule` into two types `RecordingRule` and `AlertingRule`?
+2. Do we need to add support for Remote Write Configuration per tenant? (See `ruler_remote_write_*` parameters in [limits_config](https://grafana.com/docs/loki/latest/configuration/#limits_config))
+3. Do we need to add support for the `ruler_evaluation_delay_duration`, `ruler_max_rules_per_rule_group` and `ruler_max_rule_groups_per_tenant` limits?
```

Comment:
@cyriltovena @simonswine What is your opinion on these open question?

Reply:
1. Not sure we do.
2. Could be useful at some point.
3. I guess this is more for expert type of operations, so not having at first is fine.